target/linux/ramips/mt7621/base-files/etc/board.d/02_network

```diff
@@ -42,10 +42,12 @@ ramips_setup_interfaces()
 	mikrotik,routerboard-750gr3)
 		ucidef_set_interfaces_lan_wan "lan2 lan3 lan4 lan5" "wan"
 		;;
-	ubnt,edgerouter-x|\
-	ubnt,edgerouter-x-sfp)
+	ubnt,edgerouter-x)
 		ucidef_set_interface_lan "eth0 eth1 eth2 eth3 eth4"
 		;;
+	ubnt,edgerouter-x-sfp)
+		ucidef_set_interfaces_lan_wan "eth1 eth2 eth3 eth4 eth5" "eth0"
```

Comment:
Where does the WAN port come from? There is nothing about it in the commit message.
IMO, that's a separate change and should be handled separately. (And I'd keep it all-LAN anyway).

Reply:
In the OEM Firmware and in 19.x and previous versions of the firmware, eth0 was always WAN.  Perhaps it should be in the commit message, like you stated.  But from the user perspective, I think it makes sense to keep eth0 as WAN.

Reply:
Well, from what I remember from my ER-X and from what's written in the manual, the default for eth0 is LAN with a static IP. If you then set up the device with the assistant, then there the default would be the eth0. So, to me, this sounds like all-LAN is default. That's also the reason why we changed this in https://github.com/openwrt/openwrt/commit/5acd1ed0be0d78847cd7d9d5599526f59babaf4d

So, from my perspective, the previous behavior in 19.xx has been "wrong" and the config break due to DSA is a perfect chance to make it right now.

Anyway, I think changing the eth0 LAN port back to WAN should be a separate commit, mentioning explicitly that it reverts the referenced one, and should obviously then do the same for er-x and er-x-sfp, as they have the same vendor OS/setup anyway.

Reply:
Luckily I have an extra erx-sfp which is still running OEM firmware.  I just did a reset and the results are as you mentioned.  eth0 is statically assigned.  But all of the other ports are unreachable/unconfigured.  

So if OpenWRT really wants to stay with the default state of the erx and erx-sfp, then eth0 should be on LAN and all of the other ports should be disconnected and unreachable.  Putting everything on LAN is also, in this case, not the expected behaviour or "wrong".

I am willing to prepare a separate PR with the changes either to make eth0 the WAN port with eth1-X on LAN (preferred) or to have eth0 the only port on LAN with no WAN.

Reply:
I have no option about this. I can keep it as it was or something else.

Reply:
Since what you do in this PR is just adding another port, IMO this should be just changed to
`ucidef_set_interface_lan "eth0 eth1 eth2 eth3 eth4 eth5"`

Reply:
I would say to just make ```ucidef_set_interface_lan "eth0 eth1 eth2 eth3 eth4 eth5"``` and I will submit a PR to create WAN on eth0.   Choosing which ports belong to which interface is not really related to enabling the SFP port.

Reply:
@adrianschmutzler Fixed.

Reply:
A new PR requesting a change to ucidef_set_interface_lan_wan can be found here.
#2961